pkg/consuldp/consul_dataplane.go

```diff
@@ -223,8 +223,8 @@ func (cdp *ConsulDataplane) Run(ctx context.Context) error {
 		select {
 		case <-ctx.Done():
 			doneCh <- nil
-		case <-proxy.Exited():
-			doneCh <- errors.New("envoy proxy exited unexpectedly")
+		case err := <-proxy.Exited():
+			doneCh <- err
```

Comment:
Would be nice to annotate this err if not nil so that we can know it was due to the proxy exiting with an error

Reply:
Done!